test/core/loop.wast

```diff
@@ -765,6 +765,26 @@
   ))
   "type mismatch"
 )
+(assert_invalid
+  (module (func $type-param-void-vs-num
+    (f32.const 0)
+    (loop (param i32)
+      (drop)
+      (br_if 0 (f32.const 1))
```

Comment:
I'm still not sure I understand what this is intended to test. You can remove the br_if entirely and it would still fail for the same reason. Same below.

Can you change these tests such that they do not contain instructions irrelevant to the (intended) cause of failure?